### PR TITLE
Object conversion refactor

### PR DESCRIPTION
**Pull request by tsvilans**
# Object conversion refactor

Conversion functions are written more modularly, since code was being repeated, and the creation of objects and object data was too intertwined. 

## detailed explanation
* Object conversion now happens outside of main `read_3dm()` function
* Object and object data creation are separated. Object creation is now done in one place, instead of in different places (`import_curve` and `import_render_mesh`) 
* Object data conversion functions now return the Blender object data
* **!!! This changes several functions !!!**

### CODE REVIEW

File: import_3dm/converters/__init__.py [hunk 1]
```diff
@@ -38,3 +38,25 @@
...
+#       and consolidate object-level conversion.
+
+def convert_object(context, ob, name, layer, rhinomat, scale):
+    """
+    Add a new object with given data, link to
+    collection given by layer
+    """
+
+    data = RHINO_TYPE_TO_IMPORT[ob.Geometry.ObjectType](context, ob, name, scale)
+    data.materials.append(rhinomat)
```
Comment (lfertig) [inline import_3dm/converters/__init__.py]: Maybe check for data!=None or ObjectType here, or warp it in a try statement? Importing an object of type InstanceReference is not going to import any new mesh data, but create an empty and link it to a collection as an instance. I can make the reference importer return None though..

Comment (tsvilans) [reply import_3dm/converters/__init__.py]: Good catch. I guess `InstanceReference`s are a special case.

Comment (tsvilans) [reply import_3dm/converters/__init__.py]: Ok @lfertig I added a check for null object data, and put a placeholder for handling `InstanceReference`s! Take a look whenever you have a moment

**Comment by lfertig** [comment]
looks good to me. nice to have it all tidied up a bit! once you have it merged I'll adapt my code, I have to wait for the next rhino3dm pip package anyway...

**Pull request merged**